### DISCUSSION TITLE
Setting a Typeface font family please help

### DISCUSSION THREAD

**Discussion by sisaacks** [mono/SkiaSharp#2388] [Q&A]
I downloaded RobotoRegular.ttf and I am trying to use that font in Skiasharp.
I have googled and googled and googed I am not sure what  I am doing wrong.  

```
public static SKPaint SelectedText => new SKPaint { Color = SKColors.Black, FakeBoldText = true, Style = SKPaintStyle.Stroke, StrokeWidth = .6f };

using (var resource = Assembly.GetExecutingAssembly().GetManifestResourceStream("RobotoRegular.ttf"))
{
      var result = SKTypeface.FromStream(resource);
        
     // result is not null but Typeface will always be null after assigning result
     // I am not sure what I am doing wrong
     SelectedText.Typeface = result;
}
```

**Comment by taublast** [comment]
You could register the font like in docs: https://learn.microsoft.com/en-us/dotnet/maui/user-interface/fonts?view=net-maui-7.0

Then you'd probably want to use SKTypeface.FromFamilyName(alias); and then only if (font == null || font.FamilyName != alias) you'll manually load from file
```
 using (Stream fileStream = await FileSystem.Current.OpenAppPackageFileAsync(realName))
                {
                    font = SKTypeface.FromStream(fileStream);

                    if (font == null)
                        font = SKTypeface.CreateDefault();
                }
```
The tricky part here would be that realName of the font might be different across platforms..

**Reply by hreyes** [reply]
Works For Me!

**Comment by sisaacks** [comment]
I finally figured out the issue.  Below I did similar to what you did above.  When I crated the SKPaint object I must set the Typeface at that point and time.  If I try to set the type face after creation, it does not work.  I am not sure why.  

var assembly = Assembly.GetExecutingAssembly();
            var stream = assembly.GetManifestResourceStream("MyMobileApp.Resources.Fonts.RobotoRegular.ttf");
            if (stream != null)
            {
                skTypeFace = SKTypeface.FromStream(stream);
            }

**Comment by TiDaGo** [comment]
Yup it's not working but

in MauiProgram.cs
```
.ConfigureFonts(fonts =>
{
	fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
	fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
	fonts.AddFont("DS-DIGI.ttf", "DsDigi");
});
```

In other solution/project files
`var msfont = new Microsoft.Maui.Graphics.Font("DsDigi");`
msfont is not null and very well, but
`var skfont = SKTypeface.FromFamilyName("DsDigi");`
all time null.

you can try add your ttf file to Resource/Raw
and use this code

```
using var data = await FileSystem.OpenAppPackageFileAsync("DS-DIGI.ttf");
_valueTextPaint.Typeface = SKTypeface.FromStream(data);
```

this is not a solution to the bug, but it is an option to use the font
